poetry/masonry/utils/package_include.py

```diff
@@ -33,6 +33,20 @@ def refresh(self):  # type: () -> PackageInclude
 
         return self.check_elements()
 
+    @property
+    def stub_only(self):  # type: () -> bool
+        # returns `True` if this a PEP 561 stub-only package,
+        # see [PEP 561](https://www.python.org/dev/peps/pep-0561/#stub-only-packages)
+        return self.package.endswith("-stubs") and all(
+            el.suffix == ".pyi" for el in self.elements if el.is_file()
```

Comment:
This doesn't work if there is a `py.typed` file, does it? While the PEP states "Note that for stub-only packages adding a py.typed marker is not needed since the name *-stubs is enough to indicate it is a source of typing information.", it also says "If a stub package is partial it MUST include partial\n in a top level py.typed file." So a partial stub-only package will have `py.typed` in addition to .pyi files.

Reply:
@henribru stub packages and stub-only packages are not the same. The quote you refering to applies to stub packages (`pyi` stubs included alongside the `py` modules), while stub-only packages don't include a `py.typed` file, no matter whether complete or not. Real-world examples: AFAIK neither [`numpy-stubs`](https://github.com/numpy/numpy-stubs) nor [`django-stubs`](https://github.com/typeddjango/django-stubs) are complete yet they don't include a `py.typed` marker.

Reply:
@henribru I may be wrong though - do you have an example of stub-only packages using the partial marker?

Reply:
The PEP states "Stub packages - these packages SHOULD supersede any installed inline package. They can be found at foopkg-stubs for package foopkg.", so I'm not sure that's right. The Partial Stub Packages section also describes merging stub packages and runtime packages, implying they're referring to separate packages, not ones containing both stubs and regular modules. 

Unfortunately the only example I can give are my own stub-only packages, so that doesn't count for much. I don't know if those packages don't include it because they have misunderstood the spec or I have. It's definitely not worded as clear as it should be.

Reply:
@henribru I have played with `mypy` a bit - your understanding of PEP 561 is the correct one, there is definitely a difference in `mypy` behaviour: without the partial marker, `mypy` doesn't see modules that are not covered (`error: Cannot find implementation or library stub for module`) while having the marker, `mypy` does find the module. I agree with you now - the current impl is not conform with PEP 561, will update. Thank you for a thorough and helpful review!

Reply:
Great to get that cleared up, thanks for verifying it! And thank you for making this PR, looking forward to Poetry working for my stub packages again.

Reply:
@henribru should be fixed now - I didn't add an explicit check for the marker file content since it seems that its presence doesn't affect `mypy` behaviour if the stubs are complete, it's simply ignored.